### ISSUE TITLE
Oracle - Found unparsable section

### ISSUE THREAD

**Issue by sunrutcon** [sqlfluff/sqlfluff#5627]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I'm facing some errors in when parsing some SQL query generated by ODI (Oracle Data Integrator).

[SQLParseError("Line 5, Position 1: Found unparsable section: 'LEFT JOIN (sf_tb2 st2)\\n  ON st1.id = st2...'")]

### Expected Behaviour

Expected is to work as without brackets in join.

### Observed Behaviour

I know it is kind specific, but query is working in database, so it would be nice to fix it ;D

### How to reproduce

Here is reproducaple example ( Oracle 19c database):

```SQL
-- DDL
create table sf_tb1 (id int, col01 varchar2(20 char));
create table sf_tb2 (id int, col01 varchar2(20 char));

-- DML
insert into sf_tb1 values (1, 'value1-tb1');
insert into sf_tb1 values (2, 'value2-tb1');
insert into sf_tb2 values (1, 'value1-tb2');
insert into sf_tb2 values (2, 'value2-tb2');
commit;

-- SQL
SELECT
  st2.col01 as some_name
FROM sf_tb1 st1
LEFT JOIN (sf_tb2 st2)
  ON st1.id = st2.id
where (1=1)
```

```Python
# python code
from sqlfluff.core import Linter

sql = """
SELECT
  st2.col01 as some_name
FROM sf_tb1 st1
LEFT JOIN (sf_tb2 st2) -- when bracket removed, working
  ON st1.id = st2.id
where (1=1)
"""

parsed = Linter(dialect='oracle').parse_string(sql)

print(parsed.violations)

```

### Dialect

oracle

### Version

sqlfluff-2.3.5-py3-none-any

### Configuration

I dont undestad the question.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by keraion** [comment]
This specific example now works in 3.0.0a6 pre-release and should be fixed in the 3.0.0 release.

**Comment by saulotoledo** [comment]
@keraion, @sunrutcon, I believe this issue can be closed.
The process still finishes with `[1 unfixable linting violations found]` in 3.4.2,k but this seems to be a separate issue.
